target/ppc/translate/vmx-impl.c.inc

```diff
@@ -3436,6 +3433,52 @@ TRANS_FLAGS(ALTIVEC, VAVGUB, do_vx_vavg, 0, MO_8)
 TRANS_FLAGS(ALTIVEC, VAVGUH, do_vx_vavg, 0, MO_16)
 TRANS_FLAGS(ALTIVEC, VAVGUW, do_vx_vavg, 0, MO_32)
 
+static void gen_vabsdu(unsigned vece, TCGv_vec t, TCGv_vec a, TCGv_vec b)
+{
+    tcg_gen_umax_vec(vece, t, a, b);
+    tcg_gen_umin_vec(vece, a, a, b);
+    tcg_gen_sub_vec(vece, t, t, a);
+}
```

Comment:
Would `sub(); abs()` be equivalent here?

Reply:
No because `a`, `b` and `t` are all unsigned values, so if the result of the subtraction ends up with `1` in the most significant bit the `abs()` will negate that. For an example thing of the absolute difference between `0xFF(255)` and `0x1 (1)` in a MO_8 operation should be `0xFE (254)` not 2

Reply:
oh ok, correct. Thanks!